### PR TITLE
Add rcsetup_api.rst, fix typo for rcsetup.cycler

### PR DESCRIPTION
**Pull request by petehuang**
Addresses #7223

### CODE REVIEW

File: doc/api/rcsetup_api.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+*****
```
Comment (dopplershift) [inline doc/api/rcsetup_api.rst]: The length of the `****` needs to match that of `rcsetup`; that's causing a warning from sphinx.

File: lib/matplotlib/rcsetup.py [hunk 2]
```diff
@@ -733,9 +733,9 @@ def cycler(*args, **kwargs):
     Creates a :class:`cycler.Cycler` object much like :func:`cycler.cycler`,
     but includes input validation.
 
-    cyl(arg)
-    cyl(label, itr)
-    cyl(label1=itr1[, label2=itr2[, ...]])
+    cyler(arg)
+    cyler(label, itr)
+    cyler(label1=itr1[, label2=itr2[, ...]])
```
Comment (WeatherGod) [inline lib/matplotlib/rcsetup.py]: still has a typo. `cyl` was the original name, but then we decided it was ok to collide with the `cycler` from the `cycler` package.

Comment (petehuang) [inline lib/matplotlib/rcsetup.py]: Hm, can you clarify what you mean by still having a typo? What's the exact?

Comment (dopplershift) [inline lib/matplotlib/rcsetup.py]: `cyler` should be `cycler`

**Comment by petehuang** [comment]
Ugh...bozo mistakes. I looked at `cyler` for like 15 minutes and still couldn't figure it out :frowning_face:

**Comment by WeatherGod** [comment]
The human brain is a weird place

**Comment by WeatherGod** [comment]
backported to v2.x branch via 3b48809

**Pull request merged**